### DISCUSSION TITLE
Would it be possible to make the object storage backend a pluggable mechanism?

### DISCUSSION THREAD

**Discussion by wjunLu** [juicedata/juicefs#5104] [Ideas]
Based on the current JuiceFS code, integrating a new object storage backend requires direct modifications and additions to the repository's code to handle the new object storage. 

This may result in a lengthy code integration cycle and the potential to impact existing functionality. Would it be possible to optimize the code architecture and adopt a `JuiceFS + External Object Storage Plugin` approach?

**Comment by zhijian-pro** [comment]
We have tried internally, but we expect to still be able to statically compile after using the plugin-based mechanism. However, testing has found that there is a conflict between the current plugin mechanism and static compilation, so the plan has been temporarily postponed.
